lib/tokensecurity.js

```diff
@@ -787,8 +798,16 @@ module.exports = function (app, config) {
 
       if (!token) {
         const header = req.headers.authorization
-        if (header && header.startsWith('JWT ')) {
-          token = header.substring('JWT '.length)
+        if (typeof header === 'string' && header.trim() !== '') {
+          token = header
+        }
+
+        if (typeof token === 'string' && token.includes('Bearer')) {
+          token = token.replace('Bearer', '').trim()
+        }
+
+        if (typeof token === 'string' && token.includes('JWT')) {
```

Comment:
If the token value just happens to include the string `JWT` or `Bearer` the auth will fail.

Reply:
@sbender9 no, that's what the `trim()` is for. @tkurki good point, didn't think that could be a case!

Reply:
@tkurki changed, please review.

Reply:
This may be nitpicking, but aren't we interested in removing `JWT ` and `Bearer ` WITH the trailing space from header values that have them? If somebody sends just a plain token value with no type separated by a space this code will now remove some data from the token value if the token it just happens to start with `Bearer` or `JWT`?

Reply:
@tkurki I am not sure any token would have a trailing or starting space as part of the actual token, seems very error prone to me. But indeed, current code does not support that case. Reason I went for this solution is that, if somebody has some weird error in their client code that causes the token to be sent _without_ space after `Bearer` or with multiple spaces it'll still work. 

A different way to do it would be to simply see if whatever the client sent matches the token the server knows (using `.includes(knownToken)`). That way we don't care about the prefix and any formatting errors that may be there. 

I suppose the questions should be:

* Do we want to be character-strict about the token? In this case we shouldn't remove anything, and simply check if `sendToken === Bearer knownToken`. 
* Do we actually care about the type of token? If no, we could just do the `includes()` test

Current solution sits between these two. I personally think it is fine, since it does what we want it to do: check that there is a token and that the token is fine. And we know that the node server never sends a token with beginning/trailing spaces.

Reply:
As you can see from the code it does not verify the incoming token by comparing strings. We must be able to extract the actual token value to decode it.

I would just remove the strings `'Bearer '` and `'JWT '` from the beginning of the incoming value if they are there, just like [the current code does with JWT](https://github.com/SignalK/signalk-server-node/blob/master/lib/tokensecurity.js#L614-L615). I don't like adding extra leniency for missing or multiple spaces.